taskt/Core/Automation/Commands/DataTable Commands/CreateDataTableCommand.cs

```diff
@@ -1,110 +1,98 @@
+        [InputSpecification("Enter the Column Names required for each column of data.")]
+        [SampleUsage("MyColumn || {vColumn}")]
+        [Remarks("")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
         public DataTable v_ColumnNameDataTable { get; set; }
 
-
         public CreateDataTableCommand()
         {
-            this.CommandName = "CreateDataTableCommand";
-            this.SelectionName = "Create DataTable";
-            this.CommandEnabled = true;
-            this.CustomRendering = true;
+            CommandName = "CreateDataTableCommand";
+            SelectionName = "Create DataTable";
+            CommandEnabled = true;
+            CustomRendering = true;
 
             //initialize data table
-            this.v_ColumnNameDataTable = new System.Data.DataTable
+            v_ColumnNameDataTable = new DataTable
             {
                 TableName = "ColumnNamesDataTable" + DateTime.Now.ToString("MMddyy.hhmmss")
             };
 
-            this.v_ColumnNameDataTable.Columns.Add("Column Name");
-
-
+            v_ColumnNameDataTable.Columns.Add("Column Name");
         }
 
         public override void RunCommand(object sender)
         {
-            var engine = (Core.Automation.Engine.AutomationEngineInstance)sender;
-            var dataTableName = v_DataTableName.ConvertToUserVariable(sender);
-            if (dataTableName == "")
-            {
-                dataTableName = engine.VariableList.Where(x => x.VariableName == v_DataTableName).FirstOrDefault().VariableName;
-            }
+            var engine = (AutomationEngineInstance)sender;
 
             DataTable Dt = new DataTable();
 
             foreach(DataRow rwColumnName in v_ColumnNameDataTable.Rows)
             {
-                Dt.Columns.Add(rwColumnName.Field<string>("Column Name"));
+                Dt.Columns.Add(rwColumnName.Field<string>("Column Name").ConvertToUserVariable(sender));
             }
 
-
-            
-
             //add or override existing variable
-            if (engine.VariableList.Any(f => f.VariableName == dataTableName))
+            if (engine.VariableList.Any(f => f.VariableName == v_DataTable))
             {
-                var selectedVariable = engine.VariableList.Where(f => f.VariableName == dataTableName).FirstOrDefault();
+                var selectedVariable = engine.VariableList.Where(f => f.VariableName == v_DataTable).FirstOrDefault();
                 selectedVariable.VariableValue = Dt;
             }
             else
             {
-                Script.ScriptVariable newDataTable = new Script.ScriptVariable
+                ScriptVariable newDataTable = new ScriptVariable
                 {
-                    VariableName = dataTableName,
+                    VariableName = v_DataTable,
                     VariableValue = Dt
                 };
 
                 engine.VariableList.Add(newDataTable);
```

Comment:
Please use `engine.AddVariable()` method to add this variable.

Reply:
I thought we weren't making these changes until we decide which variable manipulation method we're sticking with

Reply:
Ok done